src/javascript/binary/websocket_pages/user/account/authenticate.js

```diff
@@ -23,6 +26,192 @@ const Authenticate = (() => {
+                    const id = $e.attr('id');
+                    const type = `${($e.attr('data-type') || '').replace(/\s/g, '_').toLowerCase()}_${id.split('_')[0]}`;
+                    const $inputs = $e.parent().parent().parent().find('input[type="text"]');
+                    const file_obj = {
+                        file: e.files[0],
+                        type,
+                    };
+                    if ($inputs.length) {
+                        file_obj.id_number = $($inputs[0]).val();
+                        file_obj.exp_date = $($inputs[1]).val();
+                    }
+                    files.push(file_obj);
+                }
+            });
+            processFiles(files);
+        };
+
+        const processFiles = (files) => {
+            const promises = [];
+            const ws = BinarySocket.get();
+            const config = {
+                connection: ws,
+            };
+            const uploader = new DocumentUploader(config);
+
+            readFiles(files).then((objects) => {
+                objects.forEach(obj => promises.push(uploader.upload(obj)));
+                Promise.all(promises)
+                    .then(() => showSuccess())
+                    .catch(showError);
+            }).catch(showError);
+        };
+
+        // Returns file promise.
+        const readFiles = (files) => {
+            const promises = [];
+            files.forEach((f) => {
+                const fr = new FileReader();
+                const promise = new Promise((resolve, reject) => {
+                    fr.onload = () => {
+                        const format = (f.file.type.split('/')[1]).toUpperCase();
+                        const obj = {
+                            filename      : f.file.name,
+                            buffer        : fr.result,
+                            documentType  : f.type,
+                            documentFormat: format,
+                        };
+                        obj.documentId = f.id_number || '';
+                        obj.expirationDate = f.exp_date || '';
+                        resolve(obj);
+                    };
+
+                    fr.onerror = () => {
+                        reject(`Unable to read file ${f.file.name}`);
+                    };
+                    // Reading file.
+                    fr.readAsArrayBuffer(f.file);
+                });
+
+                promises.push(promise);
+            });
+
+            return Promise.all(promises);
+        };
+
+        const showError = (e) => {
+            const $error = $('.error-msg');
+            const message = e.message || e.message_to_client;
+            enableButton();
+            $error.removeClass('invisible').text(message);
+            setTimeout(() => { $error.empty().addClass('invisible'); }, 3000);
+        };
+
+        const showSuccess = () => {
+            $('#authentication-message > div').not('.invisible').addClass('invisible');
```

Comment:
anywhere in the code, `addClass('invisible')` can be replaced with `setVisibility(0)`